src/main/java/org/prebid/server/bidder/intertech/IntertechBidder.java

```diff
@@ -0,0 +1,229 @@
+package org.prebid.server.bidder.intertech;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.intertech.ExtImpIntertech;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class IntertechBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpIntertech>> INTERTECH_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private static final String PAGE_ID_MACRO = "{{page_id}}";
+    private static final String IMP_ID_MACRO = "{{imp_id}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public IntertechBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<HttpRequest<BidRequest>> bidRequests = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        final String referer = getReferer(request);
+        final List<String> curs = request.getCur();
+        final String cur = curs != null && !curs.isEmpty() ? curs.get(0) : "";
```

Comment:
Maybe extract this to separate method like you did with referer? (for consistency)